datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
It does indeed generate the SQL and issue it to the database, but it does so as part of a transaction:

> Database operations will be issued in the current transactional context and do not affect the state of the transaction, unless an error occurs, in which case the entire transaction is rolled back.

Check out [the committing](http://docs.sqlalchemy.org/en/latest/orm/session_basics.html#committing) documentation and the [transaction lifecycle](http://docs.sqlalchemy.org/en/latest/orm/session_transaction.html#unitofwork-transaction) documentation.

You're going to see ``IntegrityErrors`` when you commit the transaction to the database.